src/main/java/Beechat.java

```diff
@@ -0,0 +1,104 @@
+            }
+            if (start.equals("mark")) {
+                try {
+                    j = Integer.valueOf((msg.split(" "))[1]) - 1;
+                    list.get(j).mark();
+                    System.out.println("Nice! I've marked this task as done:\n" + list.get(j));
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println("Please mark in this form: \n" +
+                            "mark [index]");
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please enter a valid index" );
+                }
+                msg = sc.nextLine();
+                continue;
+            }
+            if (start.equals("unmark")) {
+                try {
+                    j = Integer.valueOf((msg.split(" "))[1]) - 1;
+                    list.get(j).unmark();
+                    System.out.println("OK, I've marked this task as not done yet:\n" + list.get(j));
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println("Please unmark in this form: \n" +
+                            "unmark [integer]");
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please enter a valid index" );
+                }
+                msg = sc.nextLine();
+                continue;
+            }
+            if (start.equals("todo")) {
+                try {
+                    String description = (msg.split("todo "))[1];
+                    task = new TodoTask(description);
+                    list.add(task);
+                    System.out.println("added: " + description);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please enter the todo task in this form: \n" +
+                            "todo [task name]");
+                }
+            }
+            if (start.equals("deadline")) {
+                try {
+                    String description = ((msg.split("deadline "))[1]).split("/by ")[0];
+                    String by = (msg.split("/by "))[1];
+                    task = new DeadlineTask(description, by);
+                    list.add(task);
+                    System.out.println("added: " + description);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please enter the deadline task in this form: \n" +
+                            "deadline [task name] /by [deadline]");
+                }
+            }
+            if (start.equals("event")) {
+                try {
+                    String description = ((msg.split("event "))[1]).split("/from ")[0];
+                    String from = (msg.split("/from "))[1].split("/to ")[0];
+                    String to = (msg.split("/to "))[1];
+                    task = new EventTask(description, from, to);
+                    list.add(task);
+                    System.out.println("added: " + description);
```

Comment:
Since there are several repeated blocks to deal with each command, perhaps you can considering create another method specially to handle the similar task, e.g. Parser, to improve code readability